packages/story-editor/src/components/fontCheck/fontCheckDialog.js

```diff
@@ -0,0 +1,196 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { TranslateWithMarkup, __ } from '@googleforcreators/i18n';
+import { trackClick } from '@googleforcreators/tracking';
+import styled from 'styled-components';
+import { useState, useCallback, useMemo } from '@googleforcreators/react';
+import {
+  Button,
+  BUTTON_SIZES,
+  BUTTON_TYPES,
+  Text,
+  THEME_CONSTANTS,
+  Icons,
+  Link,
+} from '@googleforcreators/design-system';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../app';
+import Dialog from '../dialog';
+import FontPicker from '../panels/design/textStyle/fontPicker';
+import { useConfig } from '../../app/config';
+import { TEXT_ELEMENT_DEFAULT_FONT } from '../../app/font/defaultFonts';
+
+const Spacer = styled.div`
+  margin-top: 22px !important;
+`;
+
+const ReversedIcon = styled(Icons.ChevronRightSmall)`
+  transform: rotate(180deg);
+`;
+
+const DashboardLink = styled.span`
+  display: flex;
+  color: black;
+  align-self: flex-start;
+  align-items: center;
+  font-weight: normal;
+  margin-right: 40px;
+`;
+
+export const FontCheckDialog = ({ isOpen, onClose, missingFont }) => {
+  const { dashboardLink, dashboardSettingsLink } = useConfig();
+  const defaultFont = useMemo(() => {
+    return {
+      font: {
+        id: TEXT_ELEMENT_DEFAULT_FONT?.family,
+        ...TEXT_ELEMENT_DEFAULT_FONT,
+      },
+    };
+  }, []);
+  const [suggestedFont, setSuggestedFont] = useState(defaultFont);
+  const { updateElementsByFontFamily } = useStory((state) => ({
+    updateElementsByFontFamily: state.actions.updateElementsByFontFamily,
+  }));
+
+  /**
+   * Handles updating suggested font which is passed to the FontPicker
+   *
+   * @param {Object} [evt] Font object.
+   * @return void
+   */
+  const onPushUpdate = useCallback((evt) => {
+    if (evt?.font) {
+      setSuggestedFont({
+        font: { id: evt.font.family, family: evt.font.family },
+      });
+    }
+  }, []);
+
+  const onSettingsClick = useCallback((evt) => {
+    trackClick(evt, 'click_video_optimization_settings');
+  }, []);
+
+  const updateMissingFontWithDefault = useCallback(() => {
+    updateElementsByFontFamily({
+      family: missingFont,
+      properties: { font: defaultFont.font },
+    });
+    onClose();
+  }, [defaultFont, onClose, missingFont, updateElementsByFontFamily]);
+
+  const updateMissingFontWithSelected = useCallback(() => {
+    updateElementsByFontFamily({
+      family: missingFont,
+      properties: { font: suggestedFont.font },
```

Comment:
Note that `suggestedFont` currently seems to include an `id` property (used for the dropdown?). We don't want that ID property to end up on the element.

Also, we want to update the element's whole font object (including metrics), not just the font family string.

Something to keep in mind especially for writing tests

Reply:
Sounds good -- I'll update to remove that ID and update the test to include metric props etc...